addons/misra.py

```diff
@@ -3185,6 +3187,39 @@ def misra_17_3(self, cfg):
         for w in cfg.clang_warnings:
             if w['message'].endswith('[-Wimplicit-function-declaration]'):
                 self.reportError(cppcheckdata.Location(w), 17, 3)
+        for token in cfg.tokenlist:
+            if token.str not in ["while", "if"]:
```

Comment:
Why is misra-17.3 only checked inside if/while condition ?

Reply:
it was probably to avoid some false positives. But it's not a good heuristic.

We don't assume that cppcheck users provide the include paths for standard headers. And the misra.py addon does not load std.cfg file.

Reply:
Yeah, I had issues with `__typeof__` gnu extension. And it was only reported when **inside** `if( )` condition.  

Gnu.cfg also was not helping.  On top of that I was not able to add standard headers in combination with `--project=.json`, but that's a different topic.